wwwapp/models.py

```diff
@@ -59,6 +59,28 @@ class WorkshopUserProfile(models.Model):
     def __str__(self):
         return '%s: %s, %s' % (self.year, self.user_profile, self.status)
 
+class PESELField(models.CharField):
+    """PESEL field, with checksum verification."""
+
+    def clean(self, *args, **kwargs):
+        pesel = args[0] 
+
+        # https://en.wikipedia.org/wiki/PESEL#Format
+
+        if len(pesel) != 11:
+            raise ValidationError('PESEL length is wrong ({})'.format(len(pesel)))
+        if not pesel.isdigit():
+            raise ValidationError('PESEL contains non-numbers')
+        pesel_digits = [int(digit) for digit in pesel]
+
+        checksum_mults = [1, 3, 7, 9] * 2 + [1, 3, 1]
+        if sum(x*y for x, y in zip(pesel_digits, checksum_mults)) % 10 != 0:
+            raise ValidationError('PESEL checksum is not valid')
+
```

Comment:
This isn't enough. What if someone passes a PESEL with the decoded month being equal to 14 ;) Check if the month and day looks OK.

Reply:
I might consider verifying the gender :P